src/app/api/mailMerge/route.ts

```diff
@@ -13,9 +21,16 @@ export async function POST(request: NextRequest) {
       );
     }
 
-    const result = await createMailMerge(validationResult.data);
-
-    return NextResponse.json({ _id: result._id }, { status: 201 });
+    try {
+      // Try to update
+      let res = await getMailMergeByType(validationResult.data.type);
+      res = await updateMailMerge(res._id, validationResult.data);
+      return NextResponse.json({ _id: res._id }, { status: 200 });
+    } catch {
+      // If no document found, create instead
+      const res = await createMailMerge(validationResult.data);
+      return NextResponse.json({ _id: res._id }, { status: 201 });
+    }
```

Comment:
The try block could fail in multiple places, and the errors thrown within the server funcs could be thrown for multiple reasons. I don't think we should rely on a thrown error to conclude that we need to create an entity